site-src/geps/gep-1058.md

```diff
@@ -0,0 +1,623 @@
+represent a config parameter or an array of either.Which of those it represents
+depends on the specific config setting it is being applied to and what the
+containing object supports.
+
+Conceptually, the basic the way a parent `Route` and child `Routes` are combined
+is that, an object in the parent route is added (logical AND) to **each**
+equivalent object(s) in the child route. This starts at the top layer of objects
+and is repeated, in sequence, down each layer.
+
+For any given level of config objects there may be a 1-to-many relationship. As
+an example, the settings in the parent's`RouteRule` are "added" to each of the
+child's `RouteRule`s.
+
+### Configuration Conflicts and Invalid Configurations
+
+During the reconciliation process, any conflict of configuration settings
+between the parent and child routes is considered fatal to the transaction and
+the entire Binding fails.
+
+A configuration conflict exists when the reconciliation process would result in
+an Ephemeral route configuration that would not be valid as a single, standalone
+`Route`.  This means, in concept, a user could save the generated Ephemeral
+route config to `my-foo-route.yaml`, remove the parent and child routes from the
+cluster, then load `my-foo-route.yaml` into the cluster, and the system system
+would accept and deploy it as a valid config.
+
+A config conflict may not be between config elements in the child and the
+parent. Instead, the conflict may be between config elements from two (or more)
+child routes that attach to the same `RouteRule` in teh parent route. For this
+reason, when a config conflict is detected, implementations should identify the
+config elements that conflict, including the routes that are the original source
+of each element, and provide that info to users via appropriate methods of
+notification  (e.g., logs, message in GUI, etc.).
+
+NOTE: A notification of the type just described may have security implications
+if the source routes of the conflict are in different namespaces. The
+notification might include confidential information from one of the namespaces,
+that should not be exposed to an administrative user from the namespace of one
+of the other source routes.
+
+Some specific types of objects and config parameters have unique rules for how
+the parent and child are combined. These are described below.
+
+### Hostname Settings
+
+In the current version of the standard,  `hostname` can be specified at both the
+listener and route level. When both are specified, each is treated as a partial
+FQDN and the two strings are combined at the point of overlap. In this proposal,
+that model is extended to allow `hostname` to be specified at the listener,
+parent route **and** child route levels. Specifying `hostname` at any
+combination of these levels is allowed but the `hostnames` are always combined
+in the sequence: listener, parent route, child route.  
```

Comment:
Is this an important use case? If not, would it simplify implementation if we said that child routes could not specify hostnames?

Reply:
It would simplify things, yes.

Reply:
+1, 3 way wildcard match is too much